src/formats/onenote.ts

```diff
@@ -0,0 +1,489 @@
+					if (index !== -1) {
+						this.selectedSections.splice(index, 1);
+					}
+				}
+			});
+		});
+	}
+
+	async import(progress: ProgressReporter): Promise<void> {
+		let outputFolder = await this.getOutputFolder();
+		if (!outputFolder) {
+			new Notice('Please select a location to export to.');
+			return;
+		}
+
+		for (let section of this.selectedSections) {
+			let sectionFolder: TFolder = await this.createFolders(outputFolder.path + '/' + section.displayName);
+
+			let pages: OnenotePage[] = (await (await fetch(`
+					https://graph.microsoft.com/v1.0/me/onenote/sections/${section.id}/pages?$select=id,title,createdDateTime,lastModifiedDateTime`,
+			this.requestParams)).json()).value;
+
+			pages.forEach(async (page) =>
+				this.processFile(progress, sectionFolder, await (await fetch(
+					`https://graph.microsoft.com/v1.0/me/onenote/pages/${page.id}/content?includeInkML=true`,
+					this.requestParams)).text(), page));
+		}
+	}
+
+	async processFile(progress: ProgressReporter, folder: TFolder, content: string, page: OnenotePage) {
+		try {
+			const splitContent = this.convertFormat(content);
+
+			let parsedPage: HTMLElement = this.getAllAttachments(splitContent.html);
+			parsedPage = this.styledElementToHTML(parsedPage);
+			parsedPage = this.fixTables(parsedPage);
+			parsedPage = this.convertTags(parsedPage);
+			parsedPage = this.convertInternalLinks(parsedPage);
+			parsedPage = await this.convertDrawings(parsedPage);
+
+			let mdContent = htmlToMarkdown(parsedPage).trim();
+			const fileRef = await this.saveAsMarkdownFile(folder, page.title!, mdContent);
+
+			await this.fetchAttachmentQueue(progress, fileRef);
+
+			// Add the last modified and creation time metadata
+			const writeOptions: DataWriteOptions = {
+				ctime: Date.parse(page.createdDateTime!.toString()) ||
+					   Date.parse(page.lastModifiedDateTime!.toString()) ||
+					   Date.now(),
+				mtime: Date.parse(page?.lastModifiedDateTime!.toString()) ||
+					   Date.parse(page?.createdDateTime!.toString()) ||
+					   Date.now(),
+			};
+			await this.vault.append(fileRef, '', writeOptions);
+			progress.reportNoteSuccess(page.title!);
+		}
+		catch (e) {
+			progress.reportFailed(page.title!, e);
+		}
+	}
+
+	// OneNote returns page data and inking data in one file, so we need to split them
+	convertFormat(input: string): { html: string; inkml: string } {
+		const output = { html: '', inkml: '' };
+		const boundary = input.split('\n')[0];
+		const parts = input.split(boundary);
+
+		for (let part of parts) {
+			if (part.trim() === '') continue;
+
+			let contentTypeLine = part.split('\n').find((line) => line.includes('Content-Type'));
+
+			let contentType = contentTypeLine!.split(';')[0].split(':')[1].trim();
+
+			// Extract the value from the part by removing the first two lines and then splitting by the boundary delimiter
+			let value = part
+				.split('\n')
+				.slice(2)
+				.join('\n')
+				.split(boundary)[0]
```

Comment:
`parts` was already created by splitting on `boundary`. Is this split going to do anything?